IDE Cell run documentation (#1858)
astronomer/docs: astro/cloud-ide/run-cells.md

---
sidebar_label: Run cells
title: Run cells in the Astro Cloud IDE
id: run-cells
description: Execute cells in the Cloud IDE's isolated testing environment to check your code works before running it as a DAG.
---

> Test your data pipeline as a whole or in parts using the Astro Cloud IDE's dedicated and isolated testing environment. If a cell contains code, you can run it in the IDE.

Reviewer: ```suggestion
Use the Astro Cloud IDE to test your data pipeline. If a cell contains code, you can run it in the Astro Cloud IDE.
```
I recommend following the guidance in our selected style guide and avoid the possessive form of product and feature names. 

![image](https://user-images.githubusercontent.com/104205257/217248143-9bd451d2-d465-446e-bc36-e61390b0cfe3.png)

## Run a cell

> In your pipeline editor, go to a cell and click **Run** to run the cell and check that your code works as expected. 

Reviewer: ```suggestion
1. In the Cloud UI, select a Workspace and then click **Cloud IDE**.
2. Select a project on the **Projects** page.
3. Select a pipeline in the **Pipelines** list.
4. In the pipeline editor, go to a cell and click **Run**. 
```

![Running a Cloud IDE cell](/img/cloud-ide/run-cell.png)

> After you run a cell, the **Logs** tab contains all logs generated by the cell run, including task logs, component logs, and errors. The **Results** tab appears if your cell runs successfully and contains the output of your cell.

Reviewer: Did you want to mention the stratus icons in the right pane?

![image](https://user-images.githubusercontent.com/104205257/217250260-d2b33533-702d-4c91-b4dc-6116c79619a6.png)

Author: I think it's supplementary to the core indicator, which is the status in the cell itself. It might make sense to mention these as part of the dependency/ project runs though.

![Results tab for a Cloud IDE cell run](/img/cloud-ide/view-results.png)

> When you run a cell, the Cloud IDE sends a request to an isolated worker in the Astronomer managed control plane. The worker executes your cell and returns the results to the Cloud IDE. Executing cells in the Cloud IDE is free. For more information on execution, see [Execution](/astro/cloud-ide/security.md#execution).

Author: Sorry @jlaneve can you remind me whether you're cool with us just saying "Cloud IDE" or should we be referring to it as "Astro Cloud IDE" or just "IDE" at all times?

Reviewer 2: just 'Cloud IDE' or 'IDE' is fine, yeah!

### Run cells with their dependencies

> You can run a cell with its upstream or downstream dependencies to test parts of your data pipeline that should run together. To run a cell and either its upstream or downstream dependencies, click the menu next to the **Run** button and select an option.

Reviewer: ```suggestion
You can run a cell with its upstream or downstream dependencies to test parts of your data pipeline that should run together. To run a cell and its upstream or downstream dependencies, click the menu next to **Run** and then select an option.
```
Have you submitted a request to get a hover text label added to the control?

Author: I added a note in productboard!

![Button to run a cell and its dependencies](/img/cloud-ide/run-dependencies.png)

When you run a cell with dependencies, the Cloud IDE runs the most upstream cell first and then runs each downstream cell in order. If you have data dependencies, the Cloud IDE passes data from upstream cells to downstream cells as defined in your code. The logs and results for a given cell run appear in the cell's editing window.

## Run your data pipeline 

> To test your entire data pipeline in a single run, click **Run** in the top right corner of your pipeline editing window. The Cloud IDE runs all cells with no upstream dependencies first, then runs any downstream cells based on your code.

Author: @jlaneve I wasn't able to make this work on my own IDE project. But is this essentially identical to running all cells with their dependencies. As in, is the backend execution workflow for an entire project fundamentally the same as running a cell with dependencies?

## Configure a GitHub repo for running cells

> You can run your code with additional dependencies that aren't configurable within the Cloud IDE, such as helper functions in an `include` folder, using the Astro Cloud IDE GitHub integration. The Cloud IDE loads files from your GitHub repository into the testing environment so that your cells have access to all required files. 

Reviewer: ```suggestion
You can use the Astro Cloud IDE GitHub integration to run your code with additional dependencies that aren't configurable within the Cloud IDE, such as helper functions in an `include` folder. The Astro Cloud IDE loads files from your GitHub repository into the testing environment so that your cells can access all of the required files. 
```

> 1. If you haven't already, deploy your project to GitHub. See [Deploy a project](deploy-project.md).

Author: ```suggestion
1. Deploy your Astro Cloud IDE project to GitHub if you haven't already. See [Deploy a project](deploy-project.md).
```

> 2. Open your project in the Astro Cloud IDE. In the **GitHub Repo** pane, click **Configure**.

Reviewer: Should you provide the steps for opening a project?

3. Configure the following values: 

    - **Clone GitHub repo during cell execution**: Click the toggle to allow the Cloud IDE to access your GitHub repository files when executing cells. Turn on this feature if you have cells that depend on helper files, such as helper functions in `include`.
    - **Disable auto sync in favor of manual sync**: Optional. Click the toggle to sync your repository only on a manual basis. When you click the toggle, a button appears in the pane that you can use to manually sync your GitHub repository to your Cloud IDE environment.

## Troubleshoot cell runs

> If you receive the error `Could not connect to cell execution environment` after running a cell, check the [Astro status page](https://status.astronomer.io/) to see whether the Astro control plane is currently experiencing issues. If the control plane is fully operational, contact [Astronomer support](https://cloud.astronomer.io/support) and share the error. Astronomer support might need to set up additional cloud infrastructure for the IDE to enable cell runs. 

Reviewer: ```suggestion
If the error message `Could not connect to cell execution environment` appears after running a cell, check the [Astro status page](https://status.astronomer.io/) to determine the operational status of the Astro control plane. If the control plane is operational, contact [Astronomer support](https://cloud.astronomer.io/support) and share the error. To enable cell runs, Astronomer support might need to set up additional cloud infrastructure for the IDE.
```
